config/zfs-build.m4

```diff
@@ -539,16 +538,6 @@ AC_DEFUN([ZFS_AC_DEFAULT_PACKAGE], [
 	esac
 	AC_MSG_RESULT([$initconfdir])
 	AC_SUBST(initconfdir)
-
-	AC_MSG_CHECKING([whether initramfs-tools is available])
-	if test -d /usr/share/initramfs-tools ; then
-		DEFINE_INITRAMFS='--define "_initramfs 1"'
-		AC_MSG_RESULT([yes])
-	else
-		DEFINE_INITRAMFS=''
-		AC_MSG_RESULT([no])
-	fi
-	AC_SUBST(DEFINE_INITRAMFS)
```

Comment:
Rather than try and auto-detect the need for this package perhaps we should add an `--enable-initramfs` configure option.

Reply:
The current naming is a bit misleading: the zfs-initramfs package is not needed for initramfs support. The zfs-initramfs package is specifically for initramfs-tools support, which the RPM-based distributions do not use. The RPM-based distributions use dracut for their initramfs needs, which is unaffected by my PR. Additionally, this block only controls whether the zfs-initramfs RPM package is built, it does not control whether initramfs-tools files are installed in non-RPM builds. If the zfs-initramfs RPM is going to be controlled by a new option, could it be an option that better reflects what it does? Something like `--enable-zfs-initramfs-rpm`, with a description in the `--help` output that explains that the only reason anyone should enable it is if they have ported initramfs-tool` to an RPM-based distribution?

Reply:
I see what you mean, the name definitely is misleading.  One slight complication is that I think we do still want to generate the `zfs-initramfs` deb package by default for the `make deb` target.  Since there is no native debian packing in the repository that means building and converting the rpm.  See the `deb-utils` target in `config/deb.am`.  Incidentally, this is the line which caused the CI failures on Debian and Ubuntu.

```
        pkg9=$${name}-initramfs-$${version}.$${arch}.rpm; \
```

Reply:
Couldn't we just add an extra check near line 544?
If the target is a rpm (not a deb), we shouldn't execute that part...

`if test -d /usr/share/initramfs-tools ; then`

Reply:
Ah! I did not realise that this was how the Debian packages were created. You are right, this PR interferes with that, it definitely still needs to be built there. Not sure how to handle that right now, will have a think.

Reply:
How about splitting it out into a separate target, so that `rpm-utils` never builds `zfs-initramfs`, but a new target contains the current logic for building `zfs-initramfs` if `/usr/share/initramfs-tools` exists? `deb-utils` can then depend on that new target. I have updated this PR to implement that, but assuming it works as intended, it should probably still be cleaned up a little bit before it can be merged.

Reply:
That sounds reasonable to me.  I'd just suggest adding a small comment above this check with a short explanation about the purpose of this package and when it will be enabled.